### ISSUE TITLE
Can't navigate to the same route twice in the Demo app

### ISSUE THREAD

**Issue by Aleksion** [FeatherweightLabs/FeatherweightRouter#1]
I only just cloned the project - so I will investigate myself.
I'll just keep adding whatever I stumble on if you don't mind! :)

**Comment by agentk** [comment]
Good idea. The same route twice is not working yet because the back button is not wired up to pop routes, and the new view stack is not compared to the current view stack when calculating actions.

**Comment by Aleksion** [comment]
I might have a solution ready in a few hours. Will update accordingly

**Comment by Aleksion** [comment]
I'll hack away at it tomorrow. There are some things I didn't account for.

However, the question is whether you want to encourage users NOT to use the default back button?
The problem with it is, that the only way to detect a back event is:

``` Swift
public override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)

        if (self.isMovingFromParentViewController()){
            // Your code...
            print("I am going back");   
        }
    }
```

The way to make this easy to use for the user - is to subclass uiviewController - and the equivalent in Appkit - and make users of the router use this instead of normal uiviewcontrollers. However, this bypasses your normal routing logic, since it pops the controller from the stack for you. You could notify the router of the change from in here, and have it update the path segment - but it'll need a popFromStack method rather than using the setStack method - and then it needs to NOT popFromViewController when this happens, since the viewController is already popped. 

Furthermore, the viewController needs access to the router - which get rather messy, since you instantiate the presenters before the router is instantiated. (this could be solved by dispatching events on a central store - hint hint ;) )
I haven't thought it completely through yet - but it might be much more clean to remove the back button by default - and provide a custom implementation (that looks exactly the same) to users instead.

Let me know and I'll hack away at it on the plane tomorrow :)

(I'm missing doing this with reduxKit btw :D - but I do get why it has to be decoupled completely ;) )

**Comment by agentk** [comment]
Cheers for the dismiss view controller fixes. We can refactor it in the future. The demo is working correctly now. Fixed in #6 #7.

**Comment by agentk** [comment]
I've change the location where dismiss get called from in #9 to didMoveToParentViewController because if a user only half swipes a view away and decides to cancel the action, then this is the only function that reliably gets called only after the actions completes.

Also move the dismiss calls to a protocol (DismissEmitter) to be able to remove dismiss from needing to be passed into every presenter and view controller. Path is still passed in as it likely contains critical state information like ids.
